agents/web-api/src/main/java/edu/jhuapl/dorset/agents/StockAgent.java

```diff
@@ -0,0 +1,228 @@
+import org.supercsv.prefs.CsvPreference;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import edu.jhuapl.dorset.Response;
+import edu.jhuapl.dorset.agents.AbstractAgent;
+import edu.jhuapl.dorset.agents.AgentRequest;
+import edu.jhuapl.dorset.agents.AgentResponse;
+import edu.jhuapl.dorset.http.HttpClient;
+
+public class StockAgent extends AbstractAgent {
+
+    private String baseurl = "https://www.quandl.com/api/v3/datasets/WIKI/";
+    private String apiKey;
+    private HttpClient client;
+    private HashMap<String, String> stockSymbolMap;
+
+    private static final CellProcessor[] processors = new CellProcessor[] {
+            new NotNull(), new NotNull(), new NotNull(), new Optional(),
+            new Optional(), new Optional(), new NotNull(), new Optional() };
+
+    /**
+     * Stock agent
+     *
+     * @param client  An http client object
+     * @param apiKey  A Quandl API key 
+     */
+    public StockAgent(HttpClient client, String apiKey) {
+        this.client = client;
+        this.apiKey = apiKey;
+
+        ICsvBeanReader csvBeanReader = null;
+
+        InputStream nasdaqCompanies = StockAgent.class.getClassLoader().getResourceAsStream("stockagent/NASDAQ_Companies.csv");
+
+        this.stockSymbolMap = new HashMap<String, String>();
+        try {
+            csvBeanReader = new CsvBeanReader(new BufferedReader( new InputStreamReader(nasdaqCompanies)),
+                    CsvPreference.STANDARD_PREFERENCE);
+            final String[] header = csvBeanReader.getHeader(true);
+            CompanyInfo companyInfo;
+            while ((companyInfo = csvBeanReader.read(CompanyInfo.class, header, processors)) != null) {
+                this.stockSymbolMap.put(companyInfo.getName(), companyInfo.getSymbol());
+            }
+
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        } finally {
+            if (csvBeanReader != null) {
+                try {
+                    csvBeanReader.close();
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+
+                }
+            }
+        }
+
+        InputStream nyseCompanies = StockAgent.class.getClassLoader().getResourceAsStream("stockagent/NYSE_Companies.csv");
+
+        try {
+            csvBeanReader = new CsvBeanReader(new BufferedReader( new InputStreamReader(nyseCompanies)),
+                    CsvPreference.STANDARD_PREFERENCE);
+            final String[] header = csvBeanReader.getHeader(true);
+            CompanyInfo companyInfo;
+            while ((companyInfo = csvBeanReader.read(CompanyInfo.class, header, processors)) != null) {
+
+                if (!this.stockSymbolMap.containsKey("apple")) {
+                    this.stockSymbolMap.put(companyInfo.getName(), companyInfo.getSymbol());
```

Comment:
This doesn't look like a good way to handle this (or at a minimum, this deserves a comment). Let's discuss.